pkg/sql/schemachanger/scbuild/internal/scbuildstmt/process.go

```diff
@@ -99,6 +103,14 @@ func init() {
 					statementType, checks))
 			}
 		}
+		// Fetch the statement tag using the statement tag method on the type,
+		// we can use this as a blacklist of blocked schema changes.
+		typTag, found := statementType.MethodByName("StatementTag")
```

Comment:
Why is reflection being used here? What could be compile time errors are now runtime errors.

Reply:
Mainly because we don't want to hard code tags inside the table and duplicate them again. We have the type information, so we can extract it using reflection. This is an init function, and the dev gen commands will crash if this fails pretty much, so there is no risk of shipping runtime errors.

Reply:
Where would the tags be duplicated?

When handling a `tree.Statement` instance like `tree.AlterTable` for example, you can get the tag by calling `StatementTag()` then check if there is an override value from `sql.schemachanger.disable_declarative_statement`. If there is an override then use it, otherwise use a default value (maybe put the default in a type switch).

Reply:
The problem is on the validation side. When the user specifies a set of tags in that setting, we need to confirm somehow they are sane. So, the current approach is to go over the supported statements and use the type information to extract that. Otherwise, we need to throw statement tags into the supportedStatement table or add a nil pointer for each type (instead of a reflect.Type)